### PR TITLE
[MMI] Fix Connect MMI and Deep link Flows

### PR DESCRIPTION
**Pull request by albertolive**
### When doing a transaction or signing, the deep link doesn't show up, and no error is emitted in the console.

What we should be seeing:

image

We also ensured the extension is at the home view when the deep link is visible. (Current behavior)

### Last, we improved the styles in the "Connect Custody" flow.

https://github.com/MetaMask/metamask-extension/assets/1182864/0e5ff63c-3604-412c-8237-06085a96cb0c

## AC

The deep link appears after doing an operation.

A PR with the necessary changes has been created for the team to review

## Ticket
https://consensyssoftware.atlassian.net/browse/MMI-3407

## Pre-merge author checklist

- [X] I've clearly explained:
  - [X] What problem this PR is solving
  - [X] How this problem was solved

## Pre-merge reviewer checklist

- [X] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [X] PR is linked to the appropriate GitHub issue

### CODE REVIEW

**Comment by albertolive** [comment]
@metamaskbot update-policies

File: ui/pages/confirm-transaction-base/confirm-transaction-base.component.js [hunk 1]
```diff
@@ -592,9 +592,8 @@ export default class ConfirmTransactionBase extends Component {
 
     this._removeBeforeUnload();
     updateCustomNonce('');
-    cancelTransaction(txData).then(() => {
-      history.push(mostRecentOverviewPage);
-    });
+    await cancelTransaction(txData);
+    history.push(mostRecentOverviewPage);
```
Comment (brad-decker) [inline ui/pages/confirm-transaction-base/confirm-transaction-base.component.js:596]: why was this change necessary?

File: ui/pages/confirm-transaction-base/confirm-transaction-base.component.js [hunk 2]
```diff
@@ -761,7 +761,7 @@ export default class ConfirmTransactionBase extends Component {
                   });
                 },
                 onDeepLinkShown: () => {
-                  this.props.clearConfirmTransaction();
+                  clearConfirmTransaction();
```
Comment (brad-decker) [inline ui/pages/confirm-transaction-base/confirm-transaction-base.component.js:764]: Was this change for code style or function?

**Review by brad-decker — CHANGES_REQUESTED** [review]
I haven't done a test of functionality yet just code review and just a couple questions about changes in confirm-transaction-base

File: ui/pages/confirm-transaction-base/confirm-transaction-base.component.js (hunk 1 above)
Comment (albertolive) [reply ui/pages/confirm-transaction-base/confirm-transaction-base.component.js:596]: IMHO, the modified code using await provides a clearer and more concise way of handling this asynchronous operation

File: ui/pages/confirm-transaction-base/confirm-transaction-base.component.js (hunk 2 above)
Comment (albertolive) [reply ui/pages/confirm-transaction-base/confirm-transaction-base.component.js:764]: Code style, we're destructuring the props, some lines above, so I want to maintain everything the same.

**Review by zone-live — DISMISSED** [review]

File: ui/pages/home/home.container.js [hunk 3]
```diff
@@ -138,7 +138,8 @@ const mapStateToProps = (state) => {
     selectedAddress,
     firstPermissionsRequestId,
     totalUnapprovedCount,
-    hasApprovalFlows: getApprovalFlows(state).length > 0,
+    hasApprovalFlows:
+      Array.isArray(getApprovalFlows) && getApprovalFlows(state).length > 0,
```
Comment (DDDDDanica) [inline ui/pages/home/home.container.js:142]: Can i ask under which condition `getApprovalFlows` is not an array?

Comment (albertolive) [reply ui/pages/home/home.container.js:142]: The mmi build failed because `.length` was undefined, so I had to add this check. This is unrelated at mmi, and it was introduced some weeks ago. So it was good to add that check. Let me know if there's something else I can do

**Review by NidhiKJha — DISMISSED** [review]

**Review by zone-live — DISMISSED** [review]

File: ui/pages/confirm-transaction-base/confirm-transaction-base.component.js (hunk 1 above)
Comment (brad-decker) [reply ui/pages/confirm-transaction-base/confirm-transaction-base.component.js:596]: In general we should try to avoid changes unnecessary to the PR, as there may be unintended consequences that then get traced down to this. This is part of our "Auditability" engineering principle. I'm going to approve and just leave this is a critique to apply to future prs.

File: ui/pages/home/home.container.js (hunk 3 above)
Comment (brad-decker) [reply ui/pages/home/home.container.js:142]: @jpuri I'd actually like to get your eyes on this or someone from confirmations team (if I should be tagging someone else let me know)

**Review by brad-decker — DISMISSED** [review]
There's some interesting logic changes to confirmations (approvals) that I think needs a second set of eyes on it from the confirmations team. See @DDDDDanica 's comment

File: ui/pages/home/home.container.js (hunk 3 above)
Comment (jpuri) [reply ui/pages/home/home.container.js:142]: Thanks a lot @brad-decker , yes we will ensure we review it earliest.

File: ui/pages/institutional/custody/custody.js [hunk 4]
```diff
@@ -634,7 +632,7 @@ const CustodyPage = () => {
           </Box>
         </Box>
       )}
-    </>
+    </Box>
   );
 };
```
Comment (jpuri) [inline ui/pages/institutional/custody/custody.js:638]: Possible to test cover changes in this component ?

**Review by jpuri — DISMISSED** [review]
Changes look good, just a small feedback I added to it.

File: ui/pages/institutional/custody/custody.js (hunk 4 above)
Comment (albertolive) [reply ui/pages/institutional/custody/custody.js:638]: Thanks for the feedback. I already did it with snapshots

**Review by NidhiKJha — APPROVED** [review]

File: ui/pages/institutional/custody/custody.js [hunk 5]
```diff
@@ -259,8 +259,7 @@ const CustodyPage = () => {
       if (Object.keys(connectRequestValue).length) {
         setConnectRequest(connectRequestValue);
         setCurrentJwt(
-          connectRequestValue.token ||
-            (await dispatch(mmiActions.getCustodianToken())),
+          connectRequestValue.token || dispatch(mmiActions.getCustodianToken()),
```
Comment (jpuri) [inline ui/pages/institutional/custody/custody.js:262]: A dispatch should not be required for getter function I think.

**Review by DDDDDanica — APPROVED** [review]

File: ui/pages/institutional/custody/custody.js (hunk 5 above)
Comment (albertolive) [reply ui/pages/institutional/custody/custody.js:262]: Thanks, @jpuri. I'll address, if needed, these changes in a following PR.
